src/vm/diagnosticprotocolhelper.h

```diff
@@ -8,15 +8,40 @@
 #ifdef FEATURE_PERFTRACING
 
 #include "common.h"
+#include <diagnosticsprotocol.h>
+
 
 class IpcStream;
 
+// The Miscellaneous command set is 0x01
+enum class MiscellaneousCommandId : uint8_t
+{
+    // reserved      = 0x00,
+    GenerateCoreDump = 0x01,
```

Comment:
I agree.  I'll go ahead and modify this so they are named more aptly.  Perhaps `DiagnosticCommandId`, so we can lump in David's profiler attach command as well?

Reply:
Wouldn't a DiagnosticsCommandId go with a DiagnosticsSet, and since this is the DiagnosticsServer every command should reasonably belong to the DiagnosticsSet? Personally I still think a Profiler set containing David's command and a Dump set containing Mike's command is my best guess at a proper hierarchy.


FWIW - I still have no problem eliminating the hierarchy all together and just have a single flat namespace for command numbering if you want to. I imagine the concern is that there will be a big switch statement somewhere rather than multiple smaller ones. Personally I don't find a big switch to be problematic. The alternative is that we have to pick our set categories up-front before we know how many similar commands will follow in the future or how their functionality is likely to logically cluster. There is a decent chance that we will put things into the same set that later we recognize could be separate, and vice-versa we'll probably create new sets for areas where we only ever create one command. I'm fine with that outcome too.

Reply:
That's a good point.  I think it is fair to think we could expand both Profiler and Dump related commands, so I'll break them out.